hsmodels/schemas/fields.py

```diff
@@ -745,17 +801,14 @@ class BoxSpatialReference(base_models.BaseCoverage):
     feature or raster aggregation expressed as a bounding box
     """
 
-    class Config:
-        title = 'Box Spatial Reference Metadata'
+    model_config = ConfigDict(title='Box Spatial Reference Metadata')
 
-        schema_config = {'read_only': ['type']}
-
-    type: str = Field(
+    type: Literal['box'] = Field(
         default="box",
-        const=True,
+        frozen=True,
         title="Spatial reference type",
         description="A string containing the type of spatial reference",
-        allow_mutation=False,
+        json_schema_extra={'readOnly': True},
```

Comment:
I'm pretty sure readOnly: True was added to the json schema if allow_mutation=False in pydantic 1. Why can't we do the same here and look at frozen instead of allow_mutation?

Reply:
In pydantic v1 when a field had `allow_mutation=False`, we still had to do this schema configuration:
https://github.com/hydroshare/hsmodels/blob/main/hsmodels/schemas/fields.py#L751